### PR TITLE
Src/update user feature and improve token

### PR DESCRIPTION
**Pull request by JuanCarlosMartinez4**

### CODE REVIEW

File: src/main/java/com/jalasoft/jfc/controller/UserController.java [hunk 1]
```diff
@@ -148,10 +157,11 @@
     @ApiOperation(value = "Authorization", notes = "Delete user values by Id",
             response = Response.class)
     public ResponseEntity<?> deleteById(@RequestParam("Id") int id) {
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/UserController.java:161]: It is necessary?

File: src/main/java/com/jalasoft/jfc/model/token/JsonWebToken.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+package com.jalasoft.jfc.model.token;
+
+import com.jalasoft.jfc.model.entity.UserEntity;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+import java.util.Date;
+
+public class JsonWebToken {
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/model/token/JsonWebToken.java:31]: Add class documentation

File: src/main/java/com/jalasoft/jfc/model/token/JsonWebToken.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    public static String createJwt(UserEntity userEntity) {
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/model/token/JsonWebToken.java]: Add method documentation

File: src/main/java/com/jalasoft/jfc/model/token/JsonWebToken.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        long nowMillis = System.currentTimeMillis();
+        Date now = new Date(nowMillis);
+        String TOKEN_SECRET = "at11-01-10-2020-fundacion-jala.org";
+        String token = Jwts.builder().setIssuedAt(now).signWith(SignatureAlgorithm.HS256, TOKEN_SECRET.getBytes())
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/model/token/JsonWebToken.java]: What is the lenght of this line code?

File: build.gradle [hunk 5]
```diff
@@ -51,4 +53,7 @@ dependencies {
 
     // Database connector
     runtimeOnly 'mysql:mysql-connector-java'
+
+    // Extra dependency.
```
Comment (Enrique-C) [inline build.gradle]: Review documentation

File: src/main/java/com/jalasoft/jfc/controller/UserController.java [hunk 6]
```diff
@@ -49,18 +56,16 @@
...
-        String token;
-        try {
-            token = Jwts.builder()
-                    .signWith(SignatureAlgorithm.HS256, TOKEN_SECRET.getBytes())
-                    .setSubject(userName)
-                    .claim("ROLE", "admin")
-                    .compact();
-        } catch (Exception ex) {
-            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(ex.getMessage());
+
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/UserController.java]: Delete this code line

File: src/main/java/com/jalasoft/jfc/controller/UserController.java [hunk 7]
```diff
@@ -95,16 +104,19 @@
...
-    public ResponseEntity<?> updateUser(@RequestParam int id, @RequestParam String userName, @RequestParam String
-            password, @RequestParam String rol, @RequestParam String email) {
-        try {
-            // implement user Entity.
-            return new ResponseEntity<Object>("usermodel", HttpStatus.OK);
-        } catch (Exception ex) {
-            // implement user Entity.
-            return new ResponseEntity<Object>("usermodel", HttpStatus.NOT_ACCEPTABLE);
+            response = Response.class, authorizations = { @Authorization(value="JWT") })
+    public ResponseEntity<UserEntity> updateUser(@RequestParam int id, @RequestParam String userName,
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/UserController.java:107]: Add a code line here

File: src/main/java/com/jalasoft/jfc/controller/UserController.java (hunk 1 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/com/jalasoft/jfc/controller/UserController.java:161]: yes, because it is part of CRUD operations.

**Review by Enrique-C — APPROVED** [review]

**Pull request merged**